### PR TITLE
feat(content-conformance): add no-link-in-heading rule

### PR DESCRIPTION
**Pull request by BRKalow**
🎟️ [Asana Task]()

---

## Description

Add a rule to ensure we do not add links to heading elements. This can mess with our anchor link generation, and coupled with that the anchor links can introduce accessibility challenges.

## PR Checklist 🚀

- [ ] Conduct thorough self-review.
- [ ] Add or update tests as appropriate.
- [ ] Write a useful description (above) to give reviewers appropriate context.
- [ ] Identify (in the description above) and document (add Asana tasks on [this board](https://app.asana.com/0/1100423001970639/list)) any technical debt that you're aware of, but are not addressing as part of this PR.

---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - https://app.asana.com/0/0/1203555682638509

### CODE REVIEW

File: packages/content-conformance/README.md [hunk 1]
```diff
@@ -161,3 +170,7 @@ All files that are checked are represented by `VFile` instances, with some addit
 ### Rule
 
 This package ships with a number of rules, and they can also be defined within consuming projects. Rules can define a number of executor functions that will be run on the different file types. See [Rules](#rules) for more information.
+
+```
+
+```
```
Comment (BRKalow) [inline packages/content-conformance/README.md:176]:
```suggestion
```

File: packages/content-conformance/src/rules/no-link-in-heading.js [hunk 2]
```diff
@@ -0,0 +1,41 @@
+import { visit } from 'unist-util-visit'
+
+/**
+ * Visit each node in a subtree to build a string representation
+ */
+function getStringValue(node) {
```
Comment (zchsh) [inline packages/content-conformance/src/rules/no-link-in-heading.js:6]: 💭 Very optional: I think we could use [mdast-util-to-string](https://github.com/syntax-tree/mdast-util-to-string) here, if we want! I tried it out recently in https://github.com/hashicorp/dev-portal/pull/1452 and it seemed to work as expected 👍

File: packages/content-conformance/README.md [hunk 3]
```diff
@@ -140,6 +140,15 @@ export default {
 }
 ```
 
+### Rule Messages
+
+Writing meaningful, actionable rule messages is important to ensure contributors can address issues on their own. Messages passed to `context.report()` should be written in an [active voice](https://writing.wisc.edu/handbook/style/ccs_activevoice). For example:
```
Comment (zchsh) [inline packages/content-conformance/README.md:145]: 🙌 Nice, I really dig the clarity and link here 💯

**Review by zchsh — APPROVED** [review]
✅ Reviewed and this looks good to me! Happy to clone this down and play around if that would be helpful, but there tests here look great so trusting those 🙇 💯

**Pull request merged**